lib/minitest/mock.rb

```diff
@@ -222,8 +227,10 @@ def stub name, val_or_callable, *block_args
 
     yield self
   ensure
-    metaclass.send :undef_method, name
-    metaclass.send :alias_method, name, new_name
-    metaclass.send :undef_method, new_name
+    unless metaclass.nil?
```

Comment:
I could be wrong because I am not entirely sure the `ensure` block is designed only for exceptions that we predict when we write the code —in which case it is fine—, or to make sure it remains silent and harmless, even for unpredicted exceptions.

This is not easy in our case, because most `ensure` block I see tend to undo what the method was doing because it was not complete. Whereas this one is trying to enforce parts of what the method is supposed to do and contains a few things which can break if the exception is not a predictable one.

I'll have a look at the code to have a better understanding of why the `ensure` block is there in the first place and think about it.